gattc_linux.go

```diff
@@ -257,12 +261,21 @@ func (c DeviceCharacteristic) EnableNotifications(callback func(buf []byte)) err
 			for sig := range c.property {
 				if sig.Name == "org.freedesktop.DBus.Properties.PropertiesChanged" {
 					interfaceName := sig.Body[0].(string)
-					if interfaceName != "org.bluez.GattCharacteristic1" {
+
+					switch {
+					case interfaceName == "org.bluez.Device1" && sig.Path == devicePath:
+						changes := sig.Body[1].(map[string]dbus.Variant)
+
+						if connected, ok := changes["Connected"].Value().(bool); ok && !connected {
+							c.EnableNotifications(nil)
```

Comment:
Doesn't this result in a recursive call?

Reply:
It is a recursive call. Is there an issue with doing that?

If you're concerned about an infinite loop, the callback passed to `c.EnableNotifications()` there is `nil`, which means it'll run the second case of the switch statement and not the first one where this call is located, so there can't be a loop here.

Reply:
I saw that, but it does make it harder for people to read.

Anyhow, this case could also be handled by having dev call `EnableNotifications(nil)` themselves when closing without the extra logic being added. We can add a note to the docs.

Reply:
The main problem I was trying to solve with that code is that devices can disconnect without the dev doing anything. For example, when they go out of range. That's a problem I was experiencing very frequently with my project.

Reply:
That can be handled by calling `EnableNotifications(nil)` in the `ConnectHandler` if the `connected` argument is `false`, but I personally feel that requiring a custom disconnect handler just to avoid a goroutine leak is extremely counterintuitive.

Reply:
Ah, OK. Thanks for the clarification.

One way that could be handled might be to correctly hook up the connect handler for Linux Advertisers via some property handler ala https://github.com/tinygo-org/bluetooth/pull/257 (not exactly that, but similar).

In the handler, you could call `device.EnableNotifications(nil)` or whatever is the correct handling for your application when devices disconnect.

What do you think?

Reply:
I addressed using a connect handler in my second comment (you probably didn't see it because I wrote it after my clarification). I'll definitely implement it that way if you feel this code shouldn't be included in the project, and that would solve the problem, but I believe it would be extremely counterintuitive and likely to lead to subtle leaks in people's projects in the future.

Reply:
If someone has an application that does not use property notifications at all aka a beacon, they would do nothing.

If they have an advertiser that does allow connects, they probably want to disable notifications as devices connect/disconnect.

I do realize that Advertisers that do not allow connections are not yet implemented in this package, but I am planning to work on that soon.

Mainly I am hoping to keep the implementations consistent. If ensuring that notifications are disabled is automatic on Linux, but not elsewhere, that would be a bit confusing as well? I don't know the perfect answer.

Reply:
Alright, no problem. I'll implement it in a disconnect handler then. In that case, though, I think it should be made clear in the docs that if you use notifications and don't do that, it may result in a goroutine leak.

Reply:
Well, we still need to implement connect handler for Advertiser for that to work for you, unless your application is a Central?

In any case, some additional work is still required of some kind.